dex/networks/eth/params.go

```diff
@@ -3,12 +3,149 @@
 
 package eth
 
-import "decred.org/dcrdex/dex"
-
-var UnitInfo = dex.UnitInfo{
-	AtomicUnit: "gwei",
-	Conventional: dex.Denomination{
-		Unit:             "ETH",
-		ConversionFactor: 1e9,
-	},
+import (
+	"math"
+	"math/big"
+	"time"
+
+	"decred.org/dcrdex/dex"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+const (
+	// GweiFactor is the amount of wei in one gwei.
+	GweiFactor = 1e9
+	// MaxBlockInterval is the number of seconds since the last header came
+	// in over which we consider the chain to be out of sync.
+	MaxBlockInterval = 180
+)
+
+var (
+	UnitInfo = dex.UnitInfo{
+		AtomicUnit: "gwei",
+		Conventional: dex.Denomination{
+			Unit:             "ETH",
+			ConversionFactor: 1e9,
+		},
+	}
+
+	// BigGweiFactor is the *big.Int form of the GweiFactor.
+	BigGweiFactor = big.NewInt(GweiFactor)
+
+	VersionedGases = map[uint32]*Gases{
+		0: v0Gases,
+	}
+)
+
+type Gases struct {
+	InitGas, AdditionalInitGas, RedeemGas, AdditionalRedeemGas, RefundGas uint64
+}
+
+var v0Gases = &Gases{
+	// InitGas is the amount of gas needed to initialize a single
+	// ethereum swap.
+	InitGas: 135000,
+	// AdditionalInitGas is the amount of gas needed to initialize
+	// additional swaps in the same transaction.
+	AdditionalInitGas: 113000,
+	// RedeemGas is the amount of gas it costs to redeem a swap.
+	RedeemGas: 63000,
+	// AdditionalRedeemGas is the amount of gas needed to redeem
+	// additional swaps in the same transaction.
+	AdditionalRedeemGas: 32000,
+	// RefundGas is the amount of gas it costs to refund a swap.
+	RefundGas: 43000,
+}
+
+// InitGas calculates the gas required for a batch of n inits.
+func InitGas(n int, contractVer uint32) uint64 {
+	if n == 0 {
+		return 0
+	}
+	g, ok := VersionedGases[contractVer]
+	if !ok {
+		return math.MaxUint64
+	}
+	return g.InitGas + (uint64(n)-1)*g.AdditionalInitGas
+}
+
+// RedeemGas calculates the gas required for a batch of n redemptions.
+func RedeemGas(n int, contractVer uint32) uint64 {
+	if n == 0 {
+		return 0
+	}
+	g, ok := VersionedGases[contractVer]
+	if !ok {
+		return math.MaxUint64
+	}
+	return g.RedeemGas + (uint64(n)-1)*g.AdditionalRedeemGas
+}
+
+func RefundGas(contractVer uint32) uint64 {
+	g, ok := VersionedGases[contractVer]
+	if !ok {
+		return math.MaxUint64
+	}
+	return g.RefundGas
+}
+
+// GweiToWei converts uint64 Gwei to *big.Int Wei.
+func GweiToWei(v uint64) *big.Int {
+	return new(big.Int).Mul(big.NewInt(int64(v)), BigGweiFactor)
+}
+
+// GweiToWei converts *big.Int Wei to uint64 Gwei.
+func WeiToGwei(v *big.Int) uint64 {
+	return new(big.Int).Div(v, BigGweiFactor).Uint64()
+}
```

Comment:
The original value could potentially be negative or overflow the returned uint64. There is also already a ToGwei in cointypes.go